### ISSUE TITLE
[Feature Request] Dynamic footer columns based on active footer widgets

### ISSUE THREAD

**Issue by Remzi1993** [bootscore/bootscore#778]
### Describe the feature that you'd like

Implemented by #776 

### Checklist

- [X] I have searched the issues and this feature has not been requested.

**Comment by crftwrk** [comment]
https://github.com/bootscore/bootscore/pull/776#issuecomment-2123113932

**Comment by stan1781** [comment]
I’ve now modified the structure of the footer code, which also enables the functionality to use footer widgets as required:
`<div class="row">
          <?php if (is_active_sidebar('footer-1')) : ?>
              <div class="<?= apply_filters('bootscore/class/footer/col', 'col-md-6 col-lg', 'footer-1'); ?>">
                <?php dynamic_sidebar('footer-1'); ?>
              </div>
          <?php endif; ?>

          <?php if (is_active_sidebar('footer-2')) : ?>
            <div class="<?= apply_filters('bootscore/class/footer/col', 'col-md-6 col-lg', 'footer-2'); ?>">
                <?php dynamic_sidebar('footer-2'); ?>
            </div>
          <?php endif; ?>

          <?php if (is_active_sidebar('footer-3')) : ?>
            <div class="<?= apply_filters('bootscore/class/footer/col', 'col-md-6 col-lg', 'footer-3'); ?>">
                <?php dynamic_sidebar('footer-3'); ?>
            </div>
          <?php endif; ?>

          <?php if (is_active_sidebar('footer-4')) : ?>
            <div class="<?= apply_filters('bootscore/class/footer/col', 'col-md-6 col-lg', 'footer-4'); ?>">
                <?php dynamic_sidebar('footer-4'); ?>
            </div>
          <?php endif; ?>
      </div>`

**Comment by IceflowRE** [comment]
Is there a way to use the footers with templates parts instead of widgets? Currently our website is not using any widgets and is overriding the footer.php, but updating the theme is then more time consuming.

**Comment by crftwrk** [comment]
> Is there a way to use the footers with templates parts instead of widgets?

Not yet, using template parts in header and footer is something for a major update later.

> Currently our website is not using any widgets and is overriding the footer.php, but updating the theme is then more time consuming.

If you have overridden `footer.php` in your child, updating the main-theme is hassle-free.

**Comment by crftwrk** [comment]
@seanvandermolen 

> I was going to make the same recommendation above, which is to move the sidebar checks outside the col.

Problem is here that if you are using, for example just Footer 1 and 4, cols are line-up as Footer 1 and "2". In that case you have to add an empty widget to Footer 2 and 3 to get the space back. Additionally then you have a bottom margin on this empty widgets which disturbs in mobile view.

If you have only used Footer 2 and 3,  then you don't need to add `justify-content-center` to the row, because as you wrote these cols are already centered due the empty Footer 1 and 4 col. You can use filters as well to change each col width. With this tools, its possible to create nearly every kind of footer as you like.


> If you wanted to really give more control to stay inline with not having to create template overrides in the child theme, you would also add a filter for the containing row because sometimes I do like to change how the footer elements line up.

Yes, we will add more and more filters, do you want to open a PR? Contributions are highly appreciated.

However, I'm completely open for any kind of improvements here. But we are responsible for thousands of sites, breaking changes are not allowed until v7 which comes with Bootstrap 6.

Using template-parts https://fullsiteediting.com/lessons/adding-full-site-editing-features-to-classic-themes/#h-5-add-block-based-template-parts is planned for v7 as well. This may solve this by itself.

**Comment by seanvandermolen** [comment]
Yeah, its almost the reverse uno of the shortcut of centering items using 2 and 3.  While it would be a breaking change, the fix would be filtering the row with `row justify-content-between`.  Is it intuitive... IDK does that even exist anymore? lol

Not sure I understand the bottom margin, if the whole col is gone there is no widget.  Actually, how WP is wrapping widgets broke my h-100 I used in a static html widget congaing cards for col 2 and 3.  One of my pet peeves  how WP wraps widgets but provides no way to not use wrap it or apply custom CSS to the wrapper.

Of course, there's more than one way to skin a cat, no ones going to do things the same way. The way I did it before 6.x was the template override. Thinking that anyone who's going to use filters and avoid duping templates will be using pretty basic layouts. If they want something fancy, their going to override the template anyways.

Maybe another way to think about it is a feature add:, an additional filter to disable col positions all together so you have the choice to consume the footer position space or not.  

I would love to open a PR, time allowing. I have handful of ideas to try out and share.
